buildkite/resource_pipeline.go

```diff
@@ -501,14 +501,11 @@ func (*pipelineResource) Schema(ctx context.Context, req resource.SchemaRequest,
 						Computed: true,
 						MarkdownDescription: "Whether to create builds for commits that are part part of a pull request." +
 							"Only valid for Bitbucket or GitHub using a `trigger_mode = \"code\"`",
+						PlanModifiers: []planmodifier.Bool{
+							custom_modifier.WhenRepositoryProviderIs(custom_modifier.RepositoryProviderBitbucket, custom_modifier.RepositoryProviderGitHub),
+						},
 						Validators: []validator.Bool{
-							boolvalidator.Any(
-								pipelinevalidation.WhenRepositoryProviderIs(pipelinevalidation.RepositoryProviderBitbucket),
-								boolvalidator.All(
-									pipelinevalidation.WhenRepositoryProviderIs(pipelinevalidation.RepositoryProviderGitHub),
-									boolvalidation.WhenString(path.MatchRoot("provider_settings").AtName("trigger_mode"), "code"),
-								),
-							),
+							boolvalidation.WhenString(path.MatchRoot("provider_settings").AtName("trigger_mode"), "code"),
```

Comment:
This custom validator already only enforces the `code` value when it is set. When this attribute is completely missing from `provider_settings` it doesn't check the value (there isn't one). So since its not a valid value when the repository is hosted on bitbucket, it is safe to keep here since it should not be present at all